lib/rules/dot-notation.js

```diff
@@ -70,27 +101,28 @@ module.exports = {
                                 propertyValue: JSON.stringify(node.property.value)
                             },
                             fix(fixer) {
-                                const leftBracket = sourceCode.getTokenAfter(node.object, astUtils.isOpeningBracketToken);
-                                const rightBracket = sourceCode.getLastToken(node);
-
-                                if (sourceCode.getFirstTokenBetween(leftBracket, rightBracket, { includeComments: true, filter: astUtils.isCommentToken })) {
-
-                                    // Don't perform any fixes if there are comments inside the brackets.
-                                    return null;
-                                }
-
-                                const tokenAfterProperty = sourceCode.getTokenAfter(rightBracket);
-                                const needsSpaceAfterProperty = tokenAfterProperty &&
-                                    rightBracket.range[1] === tokenAfterProperty.range[0] &&
-                                    !astUtils.canTokensBeAdjacent(String(node.property.value), tokenAfterProperty);
-
-                                const textBeforeDot = astUtils.isDecimalInteger(node.object) ? " " : "";
-                                const textAfterProperty = needsSpaceAfterProperty ? " " : "";
-
-                                return fixer.replaceTextRange(
-                                    [leftBracket.range[0], rightBracket.range[1]],
-                                    `${textBeforeDot}.${node.property.value}${textAfterProperty}`
-                                );
+                                return fixDotNotationWithValue(fixer, node, node.property.value);
+                            }
+                        });
+                    }
+                }
+                if (
+                    node.computed &&
+                    node.property.type === "TemplateLiteral" &&
+                    node.property.expressions.length === 0 &&
+                    node.property.quasis.length === 1 &&
+                    validIdentifier.test(node.property.quasis[0].value.raw) &&
+                    (allowKeywords || keywords.indexOf(String(node.property.quasis[0].value.raw)) === -1)
+                ) {
+                    if (!(allowPattern && allowPattern.test(node.property.quasis[0].value.raw))) {
```

Comment:
In general I think it would be better to use the `.cooked` property of the template literal in these cases, rather than the `.raw` property. The difference is that if a template literal contains escape sequences like `\n`, the `.cooked` property will contain the "computed" escape sequence (in this case, a newline) whereas the `.raw` property will contain the raw escape sequence that was present in the string (in this case, a backslash followed by an `n`).

In this case, using `cooked` would ensure that the rule reports an error when linting strange code like ``foo[`b\ar`]``, which is equivalent to `foo.bar`.